internal/manifestwork/manifestwork.go

```diff
@@ -0,0 +1,120 @@
+package manifestwork
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	reflect "reflect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	workv1 "open-cluster-management.io/api/work/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/constants"
+)
+
+//go:generate mockgen -source=manifestwork.go -package=manifestwork -destination=mock_manifestwork.go
+
+type ManifestWorkCreator interface {
```

Comment:
Is `Creator` the right name, considering that it also runs garbage collection?

Reply:
That is a valid point and we should definitely revisit the naming of this API, but because of the lack of a better name, let's keep the convention already found in the codebase (like with the [DaemonSetCreator](https://github.com/kubernetes-sigs/kernel-module-management/blob/main/internal/daemonset/daemonset.go#L33)) and come back to this at a later point.